### PR TITLE
Instance Demo.

### PR DESCRIPTION
**Pull request by pruidong**
The isSquareA and isSquareF methods come from the example method 1 and method 6, but some codes are removed on the original basis.

The isSquare method comes from example method 3, with the extra parentheses removed. The entire method is only one line of code. I think this is what I want.

The methods isSquare, isSquareA, and isSquareF have been tested and meet expectations.

### CODE REVIEW

File: src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return shape instanceof Square;
+    }
+
+    static boolean isSquareF(Shape shape) {
+        if (shape instanceof Rectangle rect)
+            return rect.length() == rect.width();
+        return shape instanceof Shape.Square;
+    }
+
+    /// nice.
```
Comment (XueleiFan) [inline src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java:50]: 规范的注释，还是使用两个‘/’。

File: src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    static boolean isSquare(Shape shape) {
+        return
+                /// is Square
+                shape instanceof Square ||
+                        /// is RectAngle
+                        shape instanceof Rectangle rect && rect.length == rect.width;
```
Comment (XueleiFan) [inline src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java:56]: 并列的语义，形式上也要对齐（54行，55行）。

File: src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        shape = new Shape.Circle(10);
+        System.out.println("It cannot be ture that " + shape +
+                " is a square: " + (!isSquare(shape)));
+        System.out.println(!isSquareA(shape));
+        System.out.println(!isSquareF(shape));
+    }
+
+    static boolean isSquareA(Shape shape) {
+        if (shape instanceof Rectangle rect)
```
Comment (XueleiFan) [inline src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java:39]: 条件语句后面，最好总是使用双引号，很多规范都这么要求的。

**Review by XueleiFan — CHANGES_REQUESTED** [review]

File: src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java (hunk 1 above)
Comment (pruidong) [reply src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java:50]: 好的，前段时间受Dart影响,大多数时候写三个"///".后面注意下.

File: src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java (hunk 2 above)
Comment (pruidong) [reply src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java:56]: 好的，谢谢老师.

File: src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java (hunk 3 above)
Comment (pruidong) [reply src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java:39]: 老师说的是大括号把?大部分时候都用了的,在这只是示例,所以没用双括号.

Comment (XueleiFan) [reply src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java:39]: 是大括号。

Comment (pruidong) [reply src/main/java/co/ivi/jus/instance/review/puruidong/Shape.java]: 已处理.

**Pull request merged**